src/Compilers/CSharp/Portable/Symbols/Source/CustomModifierUtils.cs

```diff
@@ -89,18 +89,20 @@ internal static TypeSymbol CopyTypeCustomModifiers(TypeSymbol sourceType, TypeSy
 
 
             // Preserve nullable modifiers as well.
-            // TODO: Do this only when feature is enabled?
-            var flagsBuilder = ArrayBuilder<bool>.GetInstance();
-            destinationType.AddNullableTransforms(flagsBuilder);
-            int position = 0;
-            int length = flagsBuilder.Count;
-            bool transformResult = resultType.ApplyNullableTransforms(flagsBuilder.ToImmutableAndFree(), ref position, out resultType);
-            Debug.Assert(transformResult && position == length);
-
-            Debug.Assert(resultType.Equals(sourceType, TypeCompareKind.IgnoreDynamicAndTupleNames)); // Same custom modifiers as source type.
-            Debug.Assert(resultType.Equals(destinationType,
-                                           TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds | // Same object/dynamic and tuple names as destination type.
-                                           TypeCompareKind.CompareNullableModifiersForReferenceTypes)); // Same nullability as destination type.
+            if (containingAssembly.Modules[0].UtilizesNullableReferenceTypes)
+            {
+                var flagsBuilder = ArrayBuilder<bool>.GetInstance();
+                destinationType.AddNullableTransforms(flagsBuilder);
+                int position = 0;
+                int length = flagsBuilder.Count;
+                bool transformResult = resultType.ApplyNullableTransforms(flagsBuilder.ToImmutableAndFree(), ref position, out resultType);
+                Debug.Assert(transformResult && position == length);
+
+                Debug.Assert(resultType.Equals(sourceType, TypeCompareKind.IgnoreDynamicAndTupleNames)); // Same custom modifiers as source type.
+                Debug.Assert(resultType.Equals(destinationType,
+                                               TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds | // Same object/dynamic and tuple names as destination type.
+                                               TypeCompareKind.CompareNullableModifiersForReferenceTypes)); // Same nullability as destination type.
+            }
```

Comment:
> } [](start = 11, length = 2)

I think we should set unknown nullability otherwise.

Reply:
Added comment.

---
In reply to: [171137508](https://github.com/dotnet/roslyn/pull/25087#discussion_r171137508) [](ancestors = 171137508)